### PR TITLE
Adding mosec

### PR DESCRIPTION
**Pull request by kemingy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kemingy** [comment]
Hi @conda-forge-admin , please ping conda-forge/help-python-c . it's ready to review

**Review by carterbox — CHANGES_REQUESTED** [review]
Probably your package has rust dependencies downloaded by cargo. Please add the licenses for all of these statically linked packages to the license section.

There are automated tools which will collect the licenses for you.

https://github.com/conda-forge/conda-forge.github.io/issues/1052#issue-609481679

**Comment by kemingy** [comment]
> Probably your package has rust dependencies downloaded by cargo. Please add the licenses for all of these statically linked packages to the license section.

Thanks for your advice. I have added a `license.json` file which contains all the direct dependencies' license declaration. The new version release already contains this file. Please take a look.

**Comment by carterbox** [comment]
@conda-forge/help-python-c, please review

File: recipes/mosec/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cargo build --release
+mkdir -p mosec/bin
+cp target/$CARGO_BUILD_TARGET/release/mosec mosec/bin
+pip install .
```
Comment (carterbox) [inline recipes/mosec/build.sh:8]:
```suggestion
$PYTHON -m pip install . --no-build-isolation \
    --no-deps --ignore-installed --no-index --no-cache-dir -vv
```

Comment (kemingy) [reply recipes/mosec/build.sh]: Thanks for your advice. I have confirmed that the latest commit can handle this. Just need to copy the binary file to the installed package directory and conda will pack everything under that directory. PTAL.

**Comment by kemingy** [comment]
Hi @conda-forge-admin , please ping conda-forge/help-python-c . it's ready to review

File: recipes/mosec/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+set -euo pipefail
+
+cargo build --release
+$PYTHON -m pip install .
+mkdir -p $SP_DIR/$PKG_NAME/bin
+cp target/$CARGO_BUILD_TARGET/release/mosec $SP_DIR/$PKG_NAME/bin
+
+# generate the license file
+cargo install cargo-license
+cargo license --direct-deps-only --authors --avoid-build-deps --avoid-dev-deps --do-not-bundle --all-features --json > license.json
```
Comment (carterbox) [inline recipes/mosec/build.sh:12]:
```suggestion
cargo license --authors --avoid-build-deps --avoid-dev-deps --do-not-bundle --all-features --json > license.json
```
Indirect dependencies are still linked in the project, so they should be credited.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/mosec/build.sh (hunk 2 above)
Comment (kemingy) [reply recipes/mosec/build.sh]: Fixed. Please take a look.

**Pull request merged**